hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,103 @@
+Not only do you have to keep up with the requirements of your domain experts, but you have deal with all the technical complexity that comes with building a language, including questions such as:
+
+- How do I get from a string to a semantic model which I can work with?
+- How do I resolve references to other parts of my model, even if they're in a separate file?
+- How do I provide a great editing experience to users of my language?
+
+This is the point where Langium comes into play. The framework aims to lower the entry point for creating a language by taking care of the technical complexity, allowing you to focus on your domain's requirements.
+
+In this chapter you'll get a closer look at the requirements programming languages usually have to implement by themselves:
+
+- [Language parsing](#language-parsing)
+- [Semantic models](#semantic-models)
+- [Cross references and linking](#cross-references-and-linking)
+- [Workspace management](#workspace-management)
+- [Editor support](#editing-support)
+
+With that in mind, Langium provides out-of-the-box solutions for these problems, with the ability to fine-tune every part of it to fit your domain requirements.
+
+---
+
+## Language Parsing
+
+Programming languages and domain specific languages (DSL) cannot be parsed using simple regular expressions (RegExp). Instead they require a more sophisticated parsing strategy. To define a custom language, you'll interact with a high level representation of your context-free grammar using the [Langium grammar language](../grammar-language), in a similar fashion to EBNF.
+
+Based on the grammar, Langium is then able to construct a parser which transforms an input string into a semantic model representation.
+
+## Semantic Models
+
+Langium grammars are not only used to parse your language, but also generate a semantic model for your Language as TypeScript interfaces, which automatically match the parsed abstract syntax tree. The following language snippet:
+
+```ts
+Person:
+    'person' // keyword 
+    name=ID // semantic assignment
+;
+```
+
+Parses a simple object `{ name: 'John' }`. To interact with this object in a type safe manner, the `langium-cli` tool generates TypeScript type definitions from its parser rule. The `Person` parser rule generates the following interface:
+
+```ts
+interface Person {
+    name: String
+}
+```
+
+These interfaces allow you to safely traverse your abstract syntax tree. In case your grammar changes, they will also notify you of any breaking changes which might require you to change your domain logic.
+
+## Cross References and Linking
+
+To express any kind of relationship between elements in your language, you will need to **reference** them.
+The process of resolving these references, i.e. identifying what element of your language hides behind a certain name, is called _linking_.
+Performing the linking process in a deterministic manner with a lot of objects in your project requires sound linking design.
```

Comment:
Just a comment, I really like this paragraph above. It flow very well, and reads quite nicely 👍 .

Reply:
Thanks, I appreciate it :)